### DISCUSSION TITLE
Create a Mesh out of a list of vertices

### DISCUSSION THREAD

**Discussion by arthuro555** [jnsmalm/pixi3d#179] [Q&A] [answered]
Hello 👋
I am trying to use rapier 3d for physics with PIXI3D, but was unable to trigger any collisions, so I tried to render its debug view.
[Rapier's debug view is a list of lines and colors](https://rapier.rs/javascript3d/classes/DebugRenderBuffers.html), and it seems to me like using a Mesh3D is the correct tool for this. However, I could not manage to get it to display anything. 

Here is how I tried to make the geometry (I also tried without indices):

```js
const debug = rapierWorld.debugRender();

// Since it is a list of lines, I assume the triangles should be p1 -> p2 -> p1
// Therefore, I generate a list of indices like [0, 1, 0, 2, 3, 2, 4, 5, 4,...]
const indices = new Uint16Array(debug.vertices.length/2)
for(let i = 0, j = 0; j<debug.vertices.length/3; i+=3, j+=2){
    indices[i]=j
    indices[i+1]=j+1
    indices[i+2]=j
}

const geometry = Object.assign(new PIXI3D.MeshGeometry3D(), {
    positions: {buffer: debug.vertices},
    indices: {buffer: indices},
    colors: {buffer: debug.colors}
});

const mesh = new PIXI3D.Mesh3D(geometry);

addPIXIObjectToStage(mesh);
```

Have I done something fundamentally wrong? Is there a better way to do this? Thanks in advance for anyone that'll help me out 🥰

**Comment by jnsmalm** [comment, marked as answer]
Hello! 

Looks like you forgot to give the mesh a material. Try:

`new PIXI3D.Mesh3D(geometry, new StandardMaterial());`

**Reply by arthuro555** [reply]
Ah, that was it, thanks! I assumed the standard material would be applied by default, since the argument was optional and I didn't think a mesh without shader would make sense...

Now I wonder, what *is* the use case of a shaderless Mesh?

**Comment by jnsmalm** [comment]
Shaderless mesh, don't remember why I made that decision actually 🥇
